src/vehicle.cpp

```diff
@@ -2987,3 +2987,27 @@ void GetVehicleSet(VehicleSet &set, Vehicle *v, uint8 num_vehicles)
 		}
 	}
 }
+
+/**
+ * Calculates the minimum power-to-weight ratio usinging the maximum weight of the ground vehicle
+ * @return power-to-weight ratio in 10ths of hp(I) per tonne
+ */
+uint32 Vehicle::GetDisplayMinPowerToWeight() const
+{
+	uint32 max_weight = 0;
+	for (const Vehicle* u = this; u != nullptr; u = u->Next())
+	{
+		const Engine *e = u->GetEngine();
+		max_weight += e->GetDisplayWeight();
+		if (!e->CanCarryCargo()) continue;
+
+		auto cargo = u->cargo_type;
+		uint32 cargo_weight = CargoSpec::Get(cargo)->weight;
+		cargo_weight *= FreightWagonMult(cargo);
+		cargo_weight *= e->DetermineCapacity(u);
+		max_weight += cargo_weight / 16;
+	}
+
+	if (max_weight == 0) return 0;
+	return GetGroundVehicleCache()->cached_power * 10u / max_weight;
```

Comment:
Doxygen header says `@return power-to-weight ratio in 10ths of hp(I) per tonne` — needs updating? (This appears to return in hp per tonne)

Reply:
This is correct, the game uses hp(I) rather than hp(M) internally.